dali/kernels/imgproc/color_manipulation/color_space_conversion_impl.cuh

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2021, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_IMGPROC_COLOR_MANIPULATION_COLOR_SPACE_CONVERSION_IMPL_H_
+#define DALI_KERNELS_IMGPROC_COLOR_MANIPULATION_COLOR_SPACE_CONVERSION_IMPL_H_
+
+#include <cuda_runtime_api.h>
+#include "dali/core/geom/vec.h"
+#include "dali/core/convert.h"
+
+namespace dali {
+namespace kernels {
+
+template <typename T>
+__inline__ __device__ T rgb_to_y(vec<3, T> rgb) {
+  return ConvertSat<T>(0.299f * rgb.x + 0.587f * rgb.y + 0.114f * rgb.z);
+}
+
+template <typename T>
+__inline__ __device__ T rgb_to_cb(vec<3, T> rgb) {
+  return ConvertSat<T>(-0.16873589f * rgb.x - 0.33126411f * rgb.y + 0.50000000f * rgb.z + 128.0f);
```

Comment:
I don't like this `+ 128.0f` here - it works well for uint8 outputs, but makes no sense for signed `T` and it's applicability is dubious with `float` output, too. Perhaps we should add half-range of `unsigned` T and don't add anything otherwise? It would affect JPEG distortion. We need to discuss this.